lighthouse-core/scripts/i18n/collection-util.js

```diff
@@ -0,0 +1,281 @@
+#!/usr/bin/env node
+/**
+* @license Copyright 2019 Google Inc. All Rights Reserved.
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+* Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+*/
+'use strict';
+
+/**
+ * @typedef ICUMessageDefn
+ * @property {string} message
+ * @property {string} [description]
+ * @property {Record<string, ICUPlaceholderDefn>} [placeholders]
+ */
+
+/**
+ * @typedef ICUPlaceholderDefn
+ * @property {string} content
+ * @property {string} [example]
+ */
+
+// @ts-ignore - @types/esprima lacks all of these
+function computeDescription(ast, property, value, startRange) {
+  const endRange = property.range[0];
+  for (const comment of ast.comments || []) {
+    if (comment.range[0] < startRange) continue;
+    if (comment.range[0] > endRange) continue;
+    if (comment.value.includes('@')) {
+      // This is a complex description with description and examples.
+      let description = '';
+      /** @type {Record<string, string>} */
+      const examples = {};
+
+      const r = /@(\w+) ({\w+})?(.*)(\n|$)/g;
+      let resArr;
+      while ((resArr = r.exec(comment.value)) !== null) {
+        const tagName = resArr[1];
+        const placeholder = resArr[2];
+        const message = resArr[3].trim();
+
+        if (tagName === 'description') {
+          description = message;
```

Comment:
maybe we should fail or something if it looks like there are non-empty lines that don't have a `@` on them after `@description`? I can see description accidentally wrapping and us forgetting to check in review

Reply:
Super valid.  I think this is a good thing to follow-up with, because currently we don't support multiline comments anyway, they turn out like this:
```javascript
    const justUIStrings =
    `const UIStrings = {
      /** Tagged description for Hello World,
       * which is a long description, that wraps. */
      message: 'Hello World',
    };`;
```
outputs:
```
Expected: "Tagged description for Hello World, which is a long description, that wraps."
Received: "Tagged description for Hello World,
           * which is a long description, that wraps."
```

So, this is already an issue.  I added a test that is set to skip for now.